multiload/linux-proc.c

```diff
@@ -252,24 +252,24 @@ GetPage (int Maximum, int data [3], LoadGraph *g)
 void
 GetMemory (int Maximum, int data [5], LoadGraph *g)
 {
-    int user, shared, buffer, cached;
+    int user, shared, cached, free;
 
     glibtop_mem mem;
 
     glibtop_get_mem (&mem);
 
     g_return_if_fail ((mem.flags & needed_mem_flags) == needed_mem_flags);
 
-    user    = rint (Maximum * (float)mem.user / (float)mem.total);
+    user    = rint (Maximum * ((float)mem.user - (float)mem.shared) / (float)mem.total);
     shared  = rint (Maximum * (float)mem.shared / (float)mem.total);
-    buffer  = rint (Maximum * (float)mem.buffer / (float)mem.total);
-    cached = rint (Maximum * (float)mem.cached / (float)mem.total);
+    cached  = rint (Maximum * ((float)mem.total - (float)mem.user - (float)mem.free) / (float)mem.total);
+    free    = rint (Maximum * (float)mem.free / (float)mem.total);
```

Comment:
There is no need to set the color of the free memory.

Reply:
There is a color picker for the free memory, so I guess it should set the color?

Reply:
The color used by free memory is the background color, you do not need a data source for free memory.
if A+B+C+D=100% then D = 100-(A+B+C) where A,B,C,D  are 0>=val and val<=100

Reply:
Do you mean "we should not pass anything as the last element in `data`"? It was passed originally, so I did not change it. Or do you mean "we should calculate `free` as `total - user - cached` ?.. I thought about it, but half of those values is already calculated as "the rest" of something.

Reply:
You only removed buffer from the sentence below:
`data [4] = Maximum-user-shared-buffer-cached;`

Reply:
So do I understand correctly that you suggest calculating `free` as `Maximum - user - cached - shared` ?

Reply:
Maximum = user + cached + shared + free?

Reply:
Yes.

(The idea is to show a "Used/Available" graph like you did, instead of trying to accurately represent the memory categories, but additionally subdivide "Used" into "Used" and "Shared", and "Available" into "Free" and "Reclaimable".)

Reply:
So I think that getting "Free" from `free` is better (and more obvious) than calculating it basically like
`total - (user - shared) - shared - (total - user - free)`